### PR TITLE
Add documentation about development infrastructure

### LINKED ISSUE #1105: Extend project development infrastructure documentation
As an external contributor I'd appreciate to have more documentation about the existing software development infrastructure.
This would help understanding how things work and what is going on.

Specific points I'm thinking about:
- About the docker container [ecpe4s/exawind-snapshot](https://hub.docker.com/r/ecpe4s/exawind-snapshot)
  - Where is it defined? / What is the Dockerfile?
  - How, when, from which version of the Dockerfile is it built? / What exactly is the workflow for building it? / Where is the ci building it defined?
- (Less important, as it can be determined by reading the ci definition) What does the ci do?
  - Which static analysis is run?
  - Does it also check the docs?
  - Which tests does the ci run and which tests does the ci not run?

### PR DESCRIPTION
**Pull request by neumantm**
Resolves #1105

### CODE REVIEW

**Comment by neumantm** [comment]
For me the bullet point list still does not render correctly.
Is that a problem on my side? Or are bullet points not supported in the nalu-wind docs?

**Comment by neumantm** [comment]
Ok when rendered by readTheDocs, the bullet points are correct (at least with my own rtd project).
(See https://neumantm-nalu-wind--2.org.readthedocs.build/en/2/source/developer/infra.html)

Seems that is either a problem with building the docs outside of rtd or with my local installation.
Could anyone check if they also have that problem?

**Review by psakievich — APPROVED** [review]

**Comment by marchdf** [comment]
@psakievich feels like this should have been merged? It is approved and all.

File: docs/source/developer/infra.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+This document describes the development infrastructure used by the nalu-wind project.
+
+Docker Container
+----------------
+
+The docker container `ecpe4s/exawind-snapshot <https://hub.docker.com/r/ecpe4s/exawind-snapshot>`_
+is build by an external collaborator in the E4S project.
+Its definition is at https://gitlab.e4s.io/uo-public/exawind-snapshot.
+
+It is based on an Ubuntu with GCC on which spack-manager is used to install exawind+hypre+openfast.
```
Comment (psakievich) [inline docs/source/developer/infra.rst:13]:
```suggestion
It is based on an Ubuntu with GCC on which exawind-manager is used to install exawind+hypre+openfast.
```

**Review by psakievich — APPROVED** [review]
I don't recall  why it wasn't merged. Found one minor tweak due to infrastructure changes.

File: docs/source/developer/infra.rst (hunk 1 above)
Comment (marchdf) [reply docs/source/developer/infra.rst:13]: I made this happen in #1277

**Pull request merged**